arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stm/TArray.kt

```diff
@@ -0,0 +1,34 @@
+package arrow.fx.coroutines.stm
+
+import arrow.fx.coroutines.STM
+
+suspend fun <A> STM.newTArray(size: Int, f: (Int) -> A): TArray<A> =
+  TArray(Array(size) { i -> TVar(f(i)) })
+suspend fun <A> STM.newTArray(size: Int, a: A): TArray<A> =
+  newTArray(size) { a }
+suspend fun <A> STM.newTArray(vararg arr: A): TArray<A> =
+  TArray(arr.map { newTVar(it) }.toTypedArray())
+suspend fun <A> STM.newTArray(xs: Iterable<A>): TArray<A> =
+  TArray(xs.map { newTVar(it) }.toTypedArray())
+
+/**
+ * A [TArray] is an array of transactional variables.
+ */
+data class TArray<A>internal constructor(internal val v: Array<TVar<A>>) {
```

Comment:
Does this need to be a `data class`? Or can this be an `inline class`? Perhaps we should put that off until it's no longer `@Experimental`?

Reply:
I had no good experience with `inlince class` yet but I'll try it :+1:

Reply:
Actually thinking about this: I really do not want the constructor exposed so inline is off the table here. I really dislike the public only way that inline classes work with

Reply:
😲 I wasn't aware of that limitation, that's a shame indeed.